src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -156389,5 +156389,82 @@ public class C
                 Diagnostic(ErrorCode.ERR_NoSuchMember, "B").WithArguments("S", "B").WithLocation(12, 24)
                 );
         }
+
+        [Fact, WorkItem(63425, "https://github.com/dotnet/roslyn/issues/63425")]
+        public void DefaultOfT()
+        {
+            var comp = CreateCompilation("""
+#nullable enable
+public class C
+{
+    public void M<T>()
+    {
+        if (default(T) is not null)
```

Comment:
Let's have a test like the following

```cs
class C<T>
{
    void M1()
    {
        if (default(T) is not null)
        {
            M2(default);
            M3(default);
        }
    }

    void M2<T>(T t) { }
    void M3<T>([DisallowNull] T t) { }
}
```

No nullable warnings should occur.

Also maybe a test which uses a form like:

```cs
class C<T>
{
    void AssertNotNull([NotNull] T t) { }

    void M1()
    {
        AssertNotNull(default);
        default(T).ToString(); // ok
    }
}
```